rosplan_planning_system/src/PlanDispatch/EsterelPlanDispatcher.cpp

```diff
@@ -3,396 +3,510 @@
+        plan_cancelled = false;
+
+        // initialise machine
+        initialise();
+
+        // begin execution
+        finished_execution = false;
+        state_changed = false;
+        bool plan_started = false;
+        while (ros::ok() && !finished_execution) {
+
+            // loop while dispatch is paused
+            while (ros::ok() && dispatch_paused) {
+                ros::spinOnce();
+                loop_rate.sleep();
+            }
+
+            // cancel plan
+            if(plan_cancelled) {
+                ROS_INFO("KCL: (%s) Plan cancelled.", ros::this_node::getName().c_str());
+                break;
+            }
+
+
+            finished_execution = true;
+            state_changed = false; 
+
+            // for each node check completion, conditions, and dispatch
+            for(std::vector<rosplan_dispatch_msgs::EsterelPlanNode>::const_iterator ci = current_plan.nodes.begin(); ci != current_plan.nodes.end(); ci++) {
+                //the main loop
+                rosplan_dispatch_msgs::EsterelPlanNode node = *ci;
+                // activate plan start edges
+                if(node.node_type == rosplan_dispatch_msgs::EsterelPlanNode::PLAN_START && !plan_started) {
+
+                        // record the time for the PLAN_START node
+                        double NOW = ros::Time::now().toSec();    
+                        node_real_dispatch_time.insert (std::pair<int,double>(node.node_id, NOW)); 
+
+                        // activate new edges
+                        std::vector<int>::const_iterator ci = node.edges_in.begin();
+                        ci = node.edges_out.begin();
+                        for(; ci != node.edges_out.end(); ci++) {
+                                edge_active[*ci] = true;
+                        }
+                        finished_execution = false;
+                        state_changed = true;
+                        plan_started = true;
+                }
+
+                                 
+                // do not check actions for nodes which are not action nodes
+                if(node.node_type != rosplan_dispatch_msgs::EsterelPlanNode::ACTION_START && node.node_type != rosplan_dispatch_msgs::EsterelPlanNode::ACTION_END)
+                    continue;
+                                
+                // If at least one node is still executing we are not done yet
+                if (action_dispatched[node.action.action_id] && !action_completed[node.action.action_id]) {
+                    finished_execution = false;
+                }
+                                
+                // check action edges
+                bool edges_activate_action = true;
+                std::vector<int>::iterator eit = node.edges_in.begin();
+                for (; eit != node.edges_in.end(); ++eit) {
+                    if(!edge_active[(*eit)]) {
+                        edges_activate_action = false;
+                        continue;
```

Comment:
This "continue" doesn't appear to do anything. Either remove it, or replace with "break".